test/test.js

```diff
@@ -59,7 +59,8 @@ function loggerTestHelper(providedOptions) {
     res: null,
     transportOptions: null,
     next: function (req, res, next) {
-      res.end('{ "message": "Hi!  I\'m a chunk!" }');
+      res.write('{ "message":');
+      res.end(' "Hi!  I\'m a chunk!" }');
```

Comment:
I think the base `next` function here should be unchanged. You can override it with your own custom `next` function inside your own test.

Something like this for the test you added:
```javascript
it('should log the entire response', function () {
  var options = {
    loggerOptions: {
      fullResponseLog: true,
      responseWhitelist: ['body']
    },
    next: function (req, res, next) {
      res.write('{ "message":');
      res.end(' "Hi!  I\'m a chunk!" }');
    }
  };
  return loggerTestHelper(options).then(function (result) {
    result.log.meta.res.should.eql({body:'{ "message": "Hi!  I\'m a chunk!" }' });
  });
});
```